src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -17,8 +17,11 @@
  * under the License.
  */
 
+import { StaticIndexPattern } from 'ui/index_patterns';
```

Comment:
Looks like `StaticIndexPattern` is a client-side library which uses the SavedObjects Public client which in turn uses the HTTP API. So it would be better if we could retrieve this directly using the server-side SavedObjects client. In addition `ui/*` imports will be deprecated after we move to the new platform so we want to keep `src/core/*` free from using this legacy code.

It's not immediately obvious to me what the best alternative would be, I'll have to dig a bit deeper into the KQL and StaticIndexPattern code.

Reply:
Maybe, we can move these types in `packages/kbn-es-query`, I will talk to @Bargs about it.

Reply:
Ah I wasn't reading the code properly, I'm fine with duplicating this type.